lib/tree/peer.go

```diff
@@ -0,0 +1,74 @@
+package tree
+
+import (
+	"github.com/dedis/cothority/lib/hashid"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/crypto/abstract"
+	"strconv"
+)
+
+// Peer represents a Cothority-member
+type Peer struct {
+	// The hostname of the peer
+	Name string
+	// The port-number
+	Port int
+	// The public-key
+	PubKey abstract.Point
+	// The private-key - mostly for ourselves
+	PrivKey abstract.Secret
+	// A network-connection - if already set up
+	Conn network.Conn
+}
+
+// PeerList regroup a number of peers in a list. One peer can be
+// member of more than one PeerList.
+type PeerList struct {
+	// A list of all peers that are part of this list
+	Peers []*Peer
+	// The hash-id of this list
+	Hash hashid.HashId
+	// The suite used in this list
+	Suite abstract.Suite
+}
+
+// NewPeerListLocalhost creates a list of n localhosts, starting
+// with port p and random keys.
+func NewPeerListLocalhost(s abstract.Suite, n, port int) *PeerList {
```

Comment:
This is also used when doing tests with `deploy`. But in fact it has to support both localhosts-only and different hosts (for deployment to deterlab). I'll change the signature to take a list of hostnames which will be used in a round-robin fashion to set up the list.